sdk/core/core/inc/az_http_pipeline.h

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_HTTP_PIPELINE_H
+#define AZ_HTTP_PIPELINE_H
+
+#include <az_contract.h>
+#include <az_http_policy.h>
+#include <az_http_request.h>
+#include <az_pair.h>
+#include <az_span.h>
+#include <az_str.h>
+
+#include <_az_cfg_prefix.h>
+
+// Start the pipeline
+AZ_NODISCARD az_result
+az_http_pipeline_process(az_http_request_builder * const hrb, az_mut_span const * const response);
```

Comment:
And we also need az_pipeline_options as an input parameter.

We can have the user_credentials inside that Options parameter.  But that options struct would also contains the retry options and some other pipeline configuration for some specific policies.  For instance, the API_version needs to be in that options struct. Then some policy will take that value and add it to the request.